pkg/broker/ingress/ingress_handler.go

```diff
@@ -318,7 +321,15 @@ func (h *Handler) receive(ctx context.Context, headers http.Header, event *cloud
 		return http.StatusBadRequest, kncloudevents.NoDuration
 	}
 
-	dispatchInfo, err := h.eventDispatcher.SendEvent(ctx, *event, *channelAddress, kncloudevents.WithHeader(headers))
+	opts := []kncloudevents.SendOption{
+		kncloudevents.WithHeader(headers),
+		kncloudevents.WithOIDCAuthentication(&types.NamespacedName{
+			Name:      "mt-broker-ingress-oidc",
+			Namespace: system.Namespace(),
+		}),
```

Comment:
@creydr should we only include this opt if OIDC is enabled?

Reply:
for sources, we only check, if the sink has an audience set. If it does, we add the JWT. Here this is done under the hood by the kncloudevents.Dispatcher: when there is a SA set and the sink has an audience -> add JWT

Reply:
Oh cool!! Thanks for clarifying